frames/src/main/java/com/checkout/frames/component/expirydate/ExpiryDateViewModel.kt

```diff
@@ -27,22 +32,68 @@ internal class ExpiryDateViewModel @Inject constructor(
     private val style: ExpiryDateComponentStyle,
 ) : ViewModel() {
 
-    val componentState = ExpiryDateComponentState(inputComponentStateMapper.map(style.inputStyle))
+    val componentState = provideViewState(style.inputStyle)
     val componentStyle = provideViewStyle(style.inputStyle)
 
+    // Flag to determine that component has been already focused before
+    // Needed to prevent validation on focus switch for initial component state
+    private var wasFocused = false
+
+    fun onFocusChanged(isFocused: Boolean) {
+        if (isFocused) wasFocused = isFocused
+
+        if (!isFocused && wasFocused) {
+            val expiryDate = componentState.expiryDate.value
+            validateExpiryDateOnFocusChanged(expiryDate)
+        }
+    }
+
+    private fun validateExpiryDateOnFocusChanged(expiryDate: String) {
+        if (expiryDate.length >= EXPIRY_DATE_MAXIMUM_LENGTH) {
```

Comment:
Compose input field has max symbols limitation.
In which scenario can expiryDate.length be more than EXPIRY_DATE_MAXIMUM_LENGTH ?

Reply:
I will improve this condition.